prepare_heatmap_data.py

```
@@ -0,0 +1,69 @@
+        for algo_idx, algo_name in enumerate(['FastHash', 'GearHash', 'MinHash']):
+            for bin_key in sorted_bins:
+                chunk_kb = bin_key / 1024.0
+                data = algo_data[algo_name].get(bin_key, {'count': 0, 'duplicates': 0})
+                if data['count'] > 0:
+                    dup_rate = data['duplicates'] / data['count']
+                else:
+                    dup_rate = 0
+                f.write(f"{chunk_kb} {algo_idx} {dup_rate}\n")
+            f.write("\n")  # Blank line between algorithms
+
+    print(f"Processed data written to cdc_heatmap_data.txt")
```

Ruff (0.14.14)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)